app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,17 @@
+class CommentsController < ApplicationController
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @post.comments.create(comments_params)
+
+    redirect_to posts_path
+  end
```

Comment:
If you are redirecting, you do not need to set @ variables - you are not accessing those variables on the page, so you can keep the scope restricted to within the create method.